views/modalTemplate.pug

```diff
@@ -0,0 +1,30 @@
+//- action: what should happen when the user confirms their action with the modal
+//- if isActionFunction is true, then action should be a function (as a string) that the button's onClick method can call
+//- otherwise action should be a route to which a form will send a post request
+mixin modalTemplate(action, title, desc, buttonText, isActionFunction)
+  div(class="modal hidden fixed z-10 inset-0 overflow-y-auto" aria-labelledby="modal-title" role="dialog" aria-modal="true")
+    div(class="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0")
+      //- modal overlay
+      div(class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" onClick='toggleModal()' aria-hidden="true")
+
+      span(class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true") &#8203;
+      //- modal
+      div(class="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full")
```

Comment:
This modal should need a dark styling too, right now the modal has a white background on dark mode which isn't the most comfortable.